lib/archethic/transaction_chain.ex

```diff
@@ -994,4 +1014,20 @@ defmodule Archethic.TransactionChain do
         {:error, :network_issue}
     end
   end
+
+  @doc """
+  Retrieve the First transaction address for a chain from P2P Quorom
+  """
+  @spec fetch_first_transaction_address_remotely(address :: binary(), nodes :: list(Node.t())) ::
+          {:ok, binary()} | {:error, :network_issue}
+  def fetch_first_transaction_address_remotely(address, nodes)
+      when is_binary(address) and is_list(nodes) do
+    case P2P.quorum_read(nodes, %GetFirstTransactionAddress{address: address}) do
```

Comment:
You may also match `{:ok, %NotFound{}}` and return an error not exists, so the caller could have different behavior if the transaction does not exists or if there is a network issue